flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/FileSourceFunction.java

```diff
@@ -123,9 +127,11 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 			OUT nextElement = serializer.createInstance();
 			nextElement =  format.nextRecord(nextElement);
 			if (nextElement == null && splitIterator.hasNext()) {
+				splitCounter.inc();
```

Comment:
I'll remove the splitCounter metric for now, and add it back in once #2100 is merged.